Microsoft.Azure.CosmosRepository/src/Providers/ICosmosClientOptionsProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.Azure.Cosmos;
+
+namespace Microsoft.Azure.CosmosRepository.Providers
+{
+    /// <summary>
+    /// The cosmos client options provider exposes a means of providing
+    /// an instance to the configured <see cref="CosmosClientOptions"/> object,
+    /// which is shared.
+    /// </summary>
+    public interface ICosmosClientOptionsProvider
+    {
+        /// <summary>
+        /// Gets the configured <see cref="CosmosClientOptions"/> instance.
+        /// </summary>
+        /// <returns>A <see cref="CosmosClientOptions"/> instance.</returns>
+        CosmosClientOptions GetClientOptions { get; }
```

Comment:
```suggestion
        CosmosClientOptions ClientOptions { get; }
```

This shouldn't be named with a "Get" prefix unless it is a function, since it is a property - just name it `ClientOptions`.